geode/src/test/java/org/apache/calcite/adapter/geode/rel/GeodeZipsTest.java

```diff
@@ -186,6 +207,103 @@ public void testItemPredicate() {
             + "      GeodeFilter(condition=[>(ITEM($2, 0), 0)])\n"
             + "        GeodeTableScan(table=[[geode, zips]])\n");
   }
+
+  @Test
+  public void testWhereWithOrForStringField() {
+    String expectedQuery = "SELECT state AS state FROM /zips "
+        + "WHERE state IN SET('RI', 'MA')";
+    calciteAssert()
+        .query("SELECT state as state "
+            + "FROM view WHERE state = 'MA' OR state = 'RI'")
+        .returnsCount(6)
+        .queryContains(
+            GeodeAssertions.query(expectedQuery));
+  }
+
+  @Test
+  public void testWhereWithOrForNumericField() {
+    calciteAssert()
+        .query("SELECT pop as pop "
+            + "FROM view WHERE pop = 34035 OR pop = 40173")
+        .returnsCount(2)
+        .queryContains(
+            GeodeAssertions.query("SELECT pop AS pop FROM /zips WHERE pop IN SET(40173, 34035)"));
+  }
+
+  @Test
+  public void testWhereWithOrForNestedNumericField() {
+    String expectedQuery = "SELECT loc[1] AS lan FROM /zips "
+        + "WHERE loc[1] IN SET(44.098538, 43.218525)";
+
+    calciteAssert()
+        .query("SELECT loc[1] as lan "
+            + "FROM view WHERE loc[1] = 43.218525 OR loc[1] = 44.098538")
+        .returnsCount(2)
+        .queryContains(
+            GeodeAssertions.query(expectedQuery));
+  }
+
+  @Test
+  public void testWhereWithOrForLargeValueList() {
+    String stateListPredicate = "state = 'IL' OR state = 'UT' OR state = 'NJ' OR state = 'AL' "
```

Comment:
This should probably be `List<String>` which is then expanded to a big string. 
Full list of states you can get dynamically directly from region.

Reply:
Fixed it